### ISSUE TITLE
diffInOpenMinutes() does not work if only one day is configured as open

### ISSUE THREAD

**Issue by janh-kramer** [spatie/opening-hours#215]
Hi there!

Thanks for this great package. 

If I am configuring the opening hours with only one date as open:

```
$openingHours = OpeningHours::create([
    'monday'     => [],
    'tuesday'    => [],
    'wednesday'  => [],
    'thursday'   => [],
    'friday'     => [],
    'saturday'   => [],
    'sunday'     => [],
    'exceptions' => [
        '2016-11-11' => ['09:00-12:00'],
    ],
]);
```

Then $openingHours->diffInOpenMinutes() is not working any more, and throws a Spatie\OpeningHours\Exceptions\MaximumLimitExceeded exeption here https://github.com/spatie/opening-hours/blob/a6b937b56d1561cdd9ebfc329bcfd8541ab7ab16/src/Helpers/DiffTrait.php#L21

The diffInSeconds function tries to find the nextOpen time for each date in the range, but since there is none nextOpen for dates after 2016-11-11, the exception is thrown. In my opinion a good fix would be to catch the exception at this point and return $time:

```
private function diffInSeconds(string $stateCheckMethod, string $nextDateMethod, string $skipDateMethod, DateTimeInterface $startDate, DateTimeInterface $endDate): float
{
    $time = 0;

    if ($endDate < $startDate) {
        return -$this->diffInSeconds($stateCheckMethod, $nextDateMethod, $skipDateMethod, $endDate, $startDate);
    }

    $date = $startDate;

    while ($date < $endDate) {
        if ($this->$stateCheckMethod($date)) {


            try {
                $date = $this->$skipDateMethod($date);
            } catch (\Spatie\OpeningHours\Exceptions\MaximumLimitExceeded $e) {
                return $time;
            }

            continue;
        }

        $nextDate = min($endDate, $this->$nextDateMethod($date));
        $time += floatval($nextDate->format('U.u')) - floatval($date->format('U.u'));
        $date = $nextDate;
    }

    return $time;
}
```

**Comment by kylekatarnls** [comment]
Catching the exception is not a solution, we could also just never throw it. But there is a good reason for it to be there, it prevents infinite loops that you would hit if you actually try to find `nextOpen()` => there is nothing to find in the future, so it would search forever, that's why there is this exception.

You can increase the limit to your convenience: `setDayLimit()`, you can try `$openingHours->setDayLimit(PHP_INT_MAX)` at your own risks.

**Comment by janh-kramer** [comment]
I fully understand why in general the exception is thrown. But here in this case, only in this function, it leads to an error as described. You cannot calculate diffInOpenMinutes, when there is only one possible date.

**Comment by kylekatarnls** [comment]
Sorry. Indeed I see your point. We can catch in `diffInSeconds` and consider that a closing time within the limit, then the whole period is open.

**Comment by janh-kramer** [comment]
In my example configuration opening hours are defined like this:
`'2016-11-11' => ['09:00-12:00']`
So when I call $openingHours->diffInOpenMinutes('2016-11-01', '2016-11-31') I expect to get 180 minutes. 
In the loop the $skipDateMethod='nextOpen' will not find anything after 2016-11-12 and therefore throw the exception, but the sum of the time is already 180 minutes. So it makes sense for me to return that time.

**Comment by kylekatarnls** [comment]
Fixed in the version [2.41.0](https://github.com/spatie/opening-hours/releases/tag/2.41.0)

**Comment by janh-kramer** [comment]
Great thanks, its working now.
